yum-centos7/swagger_server/controllers/downloads_controller.py

```diff
@@ -43,23 +51,38 @@ def _download_packages(self, patterns):
 
         yb.install(results[0])
         yb.resolveDeps()
-        all_packages = self.__get_package_list(yb)
+        all_packages = self._get_package_list(yb)
         yb.downloadPkgs(all_packages)
 
         package_local_paths = [pkg.localpath for pkg in all_packages]
         return package_local_paths
 
+    def _delete_request(self, request_id):
+        processed_request = self.__processed_requests.pop(request_id, None)
+        os.remove(processed_request.output_file_path)
+
     def _handle_request(self, request_id):
-        request = self.__processed_requests[request_id].request
-        request.status = 'Downloading'
-        package_local_paths = self._download_packages(request.spec.packages)
-        request.status = 'Compressing'
-        # TODO Compress
-        # TODO Remove original packages
-        request.status = 'Ready'
-        request.is_consumable = True
+        req = self.__processed_requests[request_id].request
+        req.status = 'Downloading'
+        package_local_paths = self._download_packages(req.spec.packages)
+        req.status = 'Compressing'
+
+        self.__processed_requests[request_id].output_file_path = self.OUTPUT_FILE_TEMPLATE.format(
+            request_id=request_id,
+            compressed_file_type=req.compressed_file_type
+        )
+
+        compress_files(req.compressed_file_type,
+                       package_local_paths,
+                       self.__processed_requests[request_id].output_file_path,
+                       password=None)
+
+        req.status = 'Ready'
+        req.is_consumable = True
+
+        self.__scheduler.enter(self.REQUEST_KEEPING_TIMEOUT_SEC, 1, self._delete_request, (request_id,))
```

Comment:
Are we sure that it is a long enough TTL?

Reply:
I believe it is for now. We can always change it later :)